doc/user-guides/acrn-dm-parameters.rst

```diff
@@ -472,3 +453,178 @@ Here are descriptions for each of these ``acrn-dm`` command line parameters:
+
+   * - ``virtio-heci``
+     - Virtio Host Embedded Controller Interface, parameters should be appended
+       with the format ``<bus>:<device>:<function>,d<0~8>``.
+
+   * - ``virtio-i2c``
+     - Virtio i2c type device,parameters with format:
+       ``<bus>[:<client_addr>[@<node>]][,<bus>[:<client_addr>[@<node>]]``
+
+       * ``<bus>`` specifies the bus number for the native I2C adapter, e.g.,
+         ``2`` means ``/dev/i2c-2``.
+       * ``<client_addr>`` specifies the address for the native client devices
+         such as ``1C`` or  ``2F``.
+       * ``@`` specifies the prefix for the ACPI node.
+       * ``<node>`` specifies the ACPI node name supported in the
+         ``acpi_node_table[]`` in the source code: only ``cam1``, ``cam2``, and
+         ``hdac`` are supported for MRB and  are platform-specific.
+
+   * - ``virtio-gpio``
+     - Virtio GPIO type device, parameters format is:
+       ``virtio-gpio,<@controller_name{offset|name[=mapping_name]:offset|name[=mapping_name]:...}@controller_name{...}...]>``
+
+       * ``controller_name``: use the command ``ls /sys/bus/gpio/devices`` to
+         check the native GPIO controller information.  Usually, the devices
+         represent the ``controller_name`` that you can use. You can also use
+         the command ``cat /sys/bus/gpio/device/XXX/dev`` to get the device id
+         that can be used to match ``/dev/XXX``, and then use ``XXX`` as the
+         ``controller_name``. On MRB and Intel NUC platforms,
+         ``controller_name`` may be ``gpiochip0``, ``gpiochip1``, ``gpiochip2``,
+         and ``gpiochip3``.
+       * ``offset|name``: use GPIO offset or its name to locate one native GPIO
+         within the GPIO controller.
+       * ``mapping_name``: is optional. If you want to use a customized name for
+         a FE GPIO, you can set a new name here.
+
+   * - ``virtio-rnd``
+     - Virtio random generater type device, with string ``kernel=on`` to
+       initialize virtio random generater state.
+
+   * - ``virtio-rpmb``
+     - Virtio Replay Protected Memory Block (RPMB) type device, with
+       ``physical_rpmb`` to specify RPMB in physical mode,
+       otherwise RPMB is in simulated mode.
+
+   * - ``virtio-audio``
+     - Virtio audio type device
+
+   * - ``virtio-net``
+     - Virtio network type device, parameter should be appended with the format:
+       ``virtio-net,<tap_name>,[mac=<XX:XX:XX:XX:XX:XX>]``.  The ``mac`` address
+       is optional, ``tap_name`` must include the keyword ``tap``.
+
+   * - ``passthru``
+     - Indicates a passthrough device. Optional parameters include:
+
+       * ``keep_gsi``: keep vGSI for MSI capable ptdev.
+       * ``no_reset``: passthrough PCI devices are reset by default when
+         assigning them to a post-launched VM. This parameter prevents this
+         reset for debugging purposes.
+       * ``d3hot_reset``: when launching a  Windows post-launched VM, this
```

Comment:
this seems such a weird option for a user to specify.  Can the build tools add this automatically if ``acrn-dm --windows`` is used?

Reply:
it is only need for passthrough device on windows guest VM

Reply:
But we could use the `is_winvm` flag in our pass-through code to detect that it's a Windows User VM starting and add this automatically, couldn't we?

Reply:
yes we could use is_winvm to do that and remove d3hot_reset parameter

Reply:
I suspect we need to leave this ``d2hot_reset`` in the documentation until the code is changed to not require it any more, right?